core/templates/components/skill-selector/select-skill-modal.component.ts

```diff
@@ -21,18 +21,22 @@ import { ConfirmOrCancelModal } from 'components/common-layout-directives/common
 import { ShortSkillSummary } from 'domain/skill/short-skill-summary.model';
 import { SkillSummaryBackendDict } from 'domain/skill/skill-summary.model';
 import { SkillsCategorizedByTopics } from 'pages/topics-and-skills-dashboard-page/skills-list/skills-list.component';
-
+export interface CategorizedSkills {
+  [topic: string]: {
+    [subtopic: string]: SkillSummaryBackendDict[];
+  };
+}
 @Component({
   selector: 'oppia-select-skill',
   templateUrl: './select-skill-modal.component.html',
 })
 export class SelectSkillModalComponent extends ConfirmOrCancelModal {
-  categorizedSkills: SkillsCategorizedByTopics;
-  skillsInSameTopicCount: number;
-  skillSummaries: SkillSummaryBackendDict[];
-  untriagedSkillSummaries: ShortSkillSummary[];
-  allowSkillsFromOtherTopics: boolean;
-  selectedSkillId: string = null;
+  categorizedSkills!: SkillsCategorizedByTopics;
+  skillsInSameTopicCount!: number;
+  skillSummaries!: SkillSummaryBackendDict[];
+  untriagedSkillSummaries!: ShortSkillSummary[];
+  allowSkillsFromOtherTopics!: boolean;
+  selectedSkillId: string | null = null;
```

Comment:
Why can this be `null`.

Reply:
I think the best way to define these properties is to use the `non-null` `(!)` assertion operator as in [[TS-7-1]](https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1) like this:

`selectedSkillId!: string;`

without initial values which I think is okay.
This is because all the properties of the `SelectSkillModalComponent` will receive values from the `SkillSelectorComponent` via `property binding` as seen in the `select-skill-modal.component.html`
Moreover these properties in the `SkillSelectorComponent` are intend gotten using the `@Input()` meaning they are retrieved from the parent and not created in there.

![image](https://user-images.githubusercontent.com/37558983/162364257-18f90ba6-e309-494e-af42-d35f9b40e768.png)

![image](https://user-images.githubusercontent.com/37558983/162366487-9f5cce0b-c2cf-4b77-928a-46422da5be7e.png)

Reply:
Checkout new changes